lib/permessage-deflate.js

```diff
@@ -377,13 +377,12 @@ class PerMessageDeflate {
       );
 
       if (fin && this.params[`${endpoint}_no_context_takeover`]) {
-        this._inflate.close();
-        this._inflate = null;
-      } else {
-        this._inflate[kTotalLength] = 0;
-        this._inflate[kBuffers] = [];
+        this._inflate.reset();
       }
 
+      this._inflate[kTotalLength] = 0;
+      this._inflate[kBuffers] = [];
```

Comment:
Nit: can you move this before the `if` statement?

Reply:
done